lego/apps/events/permissions.py

```diff
@@ -10,6 +16,38 @@ class EventPermissionHandler(PermissionHandler):
 
     perms_without_object = [CREATE, "administrate"]
 
+    def event_type_keyword_permissions(self, event_type, perm):
+        """
+        Get the keyword permission string required for a permission for a specific event type
+
+        :return: The keyword permission the user needs
+        """
+        from lego.apps.events.models import Event
+
+        return self.keyword_permission(Event, perm) + "{event_type}/".format(
+            event_type=event_type
+        )
+
+    def has_event_type_level_permission(self, user, request, perm):
+        """
+        Check if a user has the required event_type
+        permissions for a certain action.
+
+        Return `True` if the user has permission to perform the request
+        """
+        if request is None:
+            return True
+        event_type = request.data.get("event_type")
+
+        # The request might be a patch without the event_type. This should be allowed
+        if event_type is None:
+            return True
```

Comment:
So if user tries to create event without `event_type`, it is allowed (from a permission perspective)?

Reply:
Also, return `True` here, does probably not necessary mean `[the] the user has permission to perform the request`

Reply:
Hmm, yes you're correct. Even though it shouldn't be possible it's more understandable if we enforce it for edit.

Reply:
Yeah. I read this as `oh, you don't change the evnet type, ofc. you have permission to do this action. Accepted`. It is however only applicable in case user has obj perm, and do not have `/events/edit/` perm. An no people should have that perm anyways, except for people with `sudo`.

Reply:
I see. The thing is that this method is only meant to check the `event_type`. It does not care about object permissions. This method must be used in conjunction with the "default" permission handlers. As you see in the permission class used in the view. The request needs to pass **both** this and the default permissions (which enforce obj permissions).

I see the confusion though. I will specify this a bit more in the docs, and maybe in some comments so it is more understandable :+1:. And I will only do this check for `EDIT`.